### PR TITLE
Write generated file names for directory saver to stdout

### PR DESCRIPTION
**Pull request by dominiklohmann**
This allows combining the directory saver with downstream UNIX tools, e.g., `vast exec '... | write ssv to directory foo' | xargs awk ...`.

### CODE REVIEW

File: libvast/builtins/connectors/directory.cpp [hunk 1]
```diff
@@ -51,7 +51,19 @@ class plugin : public virtual saver_plugin {
...
     auto new_args = std::vector<std::string>{file_path.string()};
     // TODO: handle overwrite semantics
-    return saver_plugin_->make_saver(new_args, std::move(info), ctrl);
+    auto file_saver
+      = saver_plugin_->make_saver(new_args, std::move(info), ctrl);
+    auto guard = caf::detail::make_scope_guard([=] {
+      fmt::print(stdout, "{}\n", file_path.string());
+    });
+    if (not file_saver)
+      return std::move(file_saver.error());
```
Comment (jachris) [inline libvast/builtins/connectors/directory.cpp:60]: I think the error check should come before creating the guard.

File: libvast/builtins/connectors/directory.cpp [hunk 2]
```diff
@@ -51,7 +51,19 @@ class plugin : public virtual saver_plugin {
                       info.input_schema.make_fingerprint(), info.format);
     auto new_args = std::vector<std::string>{file_path.string()};
     // TODO: handle overwrite semantics
-    return saver_plugin_->make_saver(new_args, std::move(info), ctrl);
+    auto file_saver
+      = saver_plugin_->make_saver(new_args, std::move(info), ctrl);
+    auto guard = caf::detail::make_scope_guard([=] {
+      fmt::print(stdout, "{}\n", file_path.string());
```
Comment (jachris) [inline libvast/builtins/connectors/directory.cpp:57]: This unconditional printing means that we also print the path if the pipeline is aborted. But perhaps that's fine.

**Review by jachris — APPROVED** [review]

File: libvast/builtins/connectors/directory.cpp (hunk 2 above)
Comment (dominiklohmann) [reply libvast/builtins/connectors/directory.cpp:57]: We still create the file in that case, right?

Comment (dominiklohmann) [reply libvast/builtins/connectors/directory.cpp:60]: I added a note to the code and left the behavior as-is.

**Pull request merged**